app/uk/gov/hmrc/tai/model/domain/TaxComponentType.scala

```diff
@@ -181,10 +181,18 @@ case object JobSeekerAllowanceIncome extends TaxCodeIncomeComponentType
 case object OtherIncome extends TaxCodeIncomeComponentType
 
 object TaxComponentType{
-  implicit val formatTaxComponentType = new Format[TaxComponentType] {
-    override def reads(json: JsValue): JsSuccess[TaxComponentType] = ???
+  implicit val formatTaxComponentType: Format[TaxComponentType] = new Format[TaxComponentType] {
+    override def reads(json: JsValue): JsSuccess[TaxComponentType] =  JsSuccess(taxComponentsMap.getOrElse(json.as[String],
+      throw new IllegalArgumentException("Invalid Tax component type")))
+
     override def writes(taxComponentType: TaxComponentType) = JsString(taxComponentType.toString)
   }
+  private val taxComponentsMap: Map[String, TaxComponentType] = Map(
```

Comment:
No need for the string match here if we just pass in TaxComponentTypes from higher up